cluster_test.go

```diff
@@ -1693,20 +1693,47 @@ func (s *S) TestMaxSocketUses(c *C) {
 		time.Sleep(5e8)
 	}
 
-	// a socket was used during connection
-	numUses := 1
+	for i := 0; i < 2; i++ {
+		stats = mgo.GetStats()
+		// every 4th usage should result in a new connection
+		c.Logf("Active connections: %d", stats.SocketsAlive)
+		s := session.Copy()
+		defer s.Close()
+		c.Check(s.Ping(), IsNil)
+		time.Sleep(time.Second)
```

Comment:
Do we have an established idiom to avoid actually sleeping in unit tests? I'm not sure, my go experience is somewhat limited thus far, but it would be nice to avoid it, e.g. by dependency injecting the time source. Not sure if that's considered idiomatic though so I defer to others.